maestro-client/src/main/java/maestro/drivers/IOSDriver.kt

```diff
@@ -45,12 +51,39 @@ class IOSDriver(
 
     private var widthPixels: Int? = null
     private var heightPixels: Int? = null
+    private var appId: String? = null
+
+    private val logger by lazy { DebugLogStore.loggerFor(IOSDriver::class.java) }
 
     override fun name(): String {
         return "iOS Simulator"
     }
 
     override fun open() {
+        ensureGrpcChannel()
+        ensureXCUITestChannel()
+    }
+
+    private fun ensureXCUITestChannel() {
+        logger.info("[Start] Installing xctest ui runner on ${iosDevice.deviceId}")
+        IOSUiTestRunner.runXCTest(iosDevice.deviceId ?: throw RuntimeException("No device selected for running UI tests"))
+        logger.info("[Done] Installing xctest ui runner on ${iosDevice.deviceId}")
+
+        logger.info("[Start] Ensuring ui test runner app is launched on ${iosDevice.deviceId}")
+        IOSUiTestRunner.ensureOpen()
+        logger.info("[Done] Ensuring ui test runner app is launched on ${iosDevice.deviceId}")
+
+        logger.info("[Start] Trying to view hierarchy for ${iosDevice.deviceId}")
+        val nodes = iosDevice
+            .contentDescriptor(appId = IOSUiTestRunner.UI_TEST_RUNNER_APP_BUNDLE_ID)
+            .get()
+        MaestroTimer.retryUntilTrue(5000) {
```

Comment:
You are retrying a static piece of code here, should you call `contentDescriptor` inside of the `retry` block?

Reply:
Good catch have replaced with call to contentdescriptor